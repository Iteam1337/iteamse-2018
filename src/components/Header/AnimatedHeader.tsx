import React from 'react'
import Lottie from 'react-lottie'
import animationData from '../../images/iteam_i.json'
import styled from '../../theme'
import GridColumn from '../Grid/GridColumn'

const Wrapper = styled.div`
  height: 650px;
`

const AnimationWrapper = styled.div`
  height: 650px;
`

const GridWrapper = styled.div`
  display: grid;
  grid-template-columns: 2fr 1fr;
`

function AnimatedHeader() {
  return (
    <Wrapper>
      <GridColumn>
        <GridWrapper>
          <div>
            <h2>
              Hej! Vi är Iteam, vi digitaliserar företag genom kod, kultur och
              strategi.
            </h2>
            <p>Läs mer om våra erbjudanden eller se våra case</p>
          </div>
          <AnimationWrapper>
            <Lottie options={{ animationData }} />
          </AnimationWrapper>
        </GridWrapper>
      </GridColumn>
    </Wrapper>
  )
}

export default AnimatedHeader
