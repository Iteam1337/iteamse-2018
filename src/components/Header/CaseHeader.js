// @flow

import React from 'react'
import styled from 'styled-components'
import Navigation from './Navigation'
import { GridColumnClean } from '../Grid/GridColumn'

type Props = {
  bgColor: ?string,
  caseImage: string,
  logo: string,
  tags: Array<?string>,
}

const Wrap = GridColumnClean.extend`
  background-color: ${({ bgColor }) => bgColor};
  height: 430px;

  @media (min-width: 481px) {
    height: 700px;
  }
`

const Content = styled.div`
  display: grid;
  grid-template-rows: auto 1fr;
`

const Information = styled.div`
  align-items: center;
  align-self: flex-end;
  display: grid;
  grid-template-columns: 1fr;
  grid-column-gap: 100px;
  padding-bottom: 60px;

  @media (min-width: 1024px) {
    grid-template-columns: repeat(2, 1fr);
  }
`

const Meta = styled.div``
const ProjectImage = styled.img`
  max-width: 100%;
  vertical-align: top;
`
const Logo = styled.img`
  margin-bottom: 40px;
  max-width: 100%;
  vertical-align: top;
`

const Tags = styled.div`
  @media (min-width: 1024px) {
    max-width: 70%;
  }
`

const Tag = styled.div`
  background-color: #fff;
  border-radius: 17px;
  display: inline-block;
  margin-bottom: 10px;
  padding: 10px 30px;
  text-align: center;

  &:not(:last-child) {
    margin-right: 10px;
  }
`

const CaseImage = styled.div`
  align-items: center;
  display: none;
  height: 500px;
  justify-content: center;
  width: 500px;

  @media (min-width: 1024px) {
    display: flex;
  }
`

const CaseHeader = ({ bgColor, caseImage, logo, tags }: Props) => {
  return (
    <Wrap bgColor={bgColor} data-test="header-case">
      <Content>
        <Navigation />
        <Information>
          <Meta>
            <Logo alt="" src={logo} />
            <Tags>{tags.map(tag => <Tag key={tag}>{tag}</Tag>)}</Tags>
          </Meta>
          <CaseImage>
            <ProjectImage alt="" src={caseImage} />
          </CaseImage>
        </Information>
      </Content>
    </Wrap>
  )
}

export default CaseHeader
