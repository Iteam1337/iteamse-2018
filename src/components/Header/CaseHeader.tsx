import React from 'react'
import styled from 'styled-components'
import { withProps } from '../../theme'
import { GridColumnClean } from '../Grid/GridColumn'
import Navigation from './Navigation'

interface CaseHeaderProps {
  bgColor: string | null
  caseImage: string | null
  logo: string
  tags: Array<string | null>
}

interface WrapProps {
  bgColor: string | null
}

const Wrap = withProps<WrapProps>()(GridColumnClean.extend)`
  background-color: ${({ bgColor }) => bgColor};
  height: 430px;

  @media (min-width: 481px) {
    height: 700px;
  }
`

const Content = styled.div`
  display: grid;
  grid-template-rows: auto 1fr;
`

const Information = styled.div`
  align-items: center;
  align-self: flex-end;
  display: grid;
  grid-template-columns: 1fr;
  grid-column-gap: 100px;
  padding-bottom: 60px;

  @media (min-width: 1025px) {
    grid-template-columns: repeat(2, 1fr);
  }
`

const Meta = styled.div``
const ProjectImage = styled.img`
  max-width: 100%;
  vertical-align: top;
`
const Logo = styled.img`
  margin-bottom: 40px;
  max-width: 100%;
  vertical-align: top;
`

const Tags = styled.div`
  @media (min-width: 1025px) {
    max-width: 70%;
  }
`

const Tag = styled.div`
  background-color: rgba(255, 255, 255, 0.15);
  color: #fff;
  display: inline-block;
  font-size: 16px;
  margin-bottom: 10px;
  padding: 10px 30px;
  text-align: center;

  &:not(:last-child) {
    margin-right: 10px;
  }
`

const CaseImage = styled.div`
  align-items: center;
  display: none;
  height: 500px;
  justify-content: center;
  width: 500px;

  @media (min-width: 1025px) {
    display: flex;
  }
`

const CaseHeader: React.SFC<CaseHeaderProps> = ({
  bgColor,
  caseImage,
  logo,
  tags,
}) => {
  return (
    <Wrap bgColor={bgColor} data-test="header-case">
      <Content>
        <Navigation />
        <Information>
          <Meta>
            <Logo alt="" src={logo} />
            <Tags>
              {tags.map(
                tag =>
                  typeof tag === 'string' ? <Tag key={tag}>{tag}</Tag> : null
              )}
            </Tags>
          </Meta>
          <CaseImage>
            {caseImage && <ProjectImage alt="" src={caseImage} />}
          </CaseImage>
        </Information>
      </Content>
    </Wrap>
  )
}

export default CaseHeader
