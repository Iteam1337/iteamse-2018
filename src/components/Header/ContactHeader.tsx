import L from 'leaflet'
import React from 'react'
import LazyLoad from 'react-lazyload'
import { Map, Marker, TileLayer } from 'react-leaflet'
import styled from '../../theme'
import { SecondaryButton } from '../Button/Button'
import { GridColumnClean } from '../Grid/GridColumn'
import Navigation from './Navigation'

const Wrap = styled(GridColumnClean)`
  background-color: ${({ theme }) => theme.colors.emperor};
  height: 430px;
  position: relative;

  @media (min-width: 481px) {
    height: 700px;
  }
`

export const Content = styled.div`
  display: grid;
  grid-column: -1 / 1;
  grid-template-rows: auto 1fr;
  grid-template-columns: 30px 1fr 30px;
  z-index: 2;

  @media (min-width: 1025px) {
    grid-template-columns: 1fr 1024px 1fr;
  }
`

const LocationNavigation = styled.div`
  align-self: flex-end;
  grid-column: 2;
  grid-row: 2;
  padding-bottom: 30px;

  @media (min-width: 481px) {
    padding-bottom: 40px;
  }
`

const MapWrapper = styled(Map)`
  bottom: 0;
  grid-column: 1 / -1;
  left: 0;
  position: absolute;
  right: 0;
  top: 0;
  z-index: 1;
`

const StyledTileLayer = styled(TileLayer)`
  filter: grayscale(100%);
`

const LocationButton =
  styled(SecondaryButton) <
  { isActive: boolean } >
  `
  background-color: ${({ theme, isActive }) =>
    isActive ? theme.colors.cornflowerBlue : theme.colors.alabaster};
  border-color: #E8E8E8;
  border-radius: 0;
  color: ${({ isActive }) => (isActive ? '#ffffff' : '#000000')};
  font-size: 14px;
  padding: 10px 24px;

  &:first-child {
    border-radius: 3px 0 0 3px;
  }

  &:last-child {
    border-radius: 0 3px 3px 0;
  }

  @media (min-width: 481px) {
    font-size: 16px;
    padding: 14px 34px;
  }
`

const markerIcon = L.icon({
  iconSize: L.point(60, 73),
  iconUrl: require('./img/marker-icon.png'),
})

const LocationMap = ({ coords }: { coords: [number, number] }) => (
  <MapWrapper center={coords} zoom={18} zoomControl={false}>
    <StyledTileLayer url="https://{s}.tile.openstreetmap.org/{z}/{x}/{y}.png" />
    <Marker icon={markerIcon} position={coords} />
  </MapWrapper>
)

type Address = {
  address1: string
  city: string
  contactMail: string
  contactPhone: string
  mapLocation: {
    lat: number
    lon: number
  }
  title: string
  zip: string
}

interface ContactHeaderProps {
  addresses: Address[]
}

interface ContactHeaderState {
  currentLocation: Address
}

class Header extends React.Component<ContactHeaderProps, ContactHeaderState> {
  state = {
    currentLocation: this.props.addresses[0],
  }

  setLocation = (location: Address) =>
    this.setState({
      currentLocation: location,
    })

  render() {
    const { currentLocation } = this.state

    return (
      <LazyLoad height={700} once>
        <Wrap>
          <LocationMap
            coords={[
              currentLocation.mapLocation.lat,
              currentLocation.mapLocation.lon,
            ]}
          />
          <Content>
            <Navigation isInverted noShadow />
            <LocationNavigation>
              {this.props.addresses.map(address => (
                <LocationButton
                  key={address.city}
                  isActive={currentLocation.city === address.city}
                  onClick={() => this.setLocation(address)}
                >
                  {address.city}
                </LocationButton>
              ))}
            </LocationNavigation>
          </Content>
        </Wrap>
      </LazyLoad>
    )
  }
}

export default Header
