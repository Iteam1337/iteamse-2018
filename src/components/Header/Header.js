// @flow

import React from 'react'
import styled from 'styled-components'
import Navigation from './Navigation'
import H1 from '../Typography/H1'
import LazyLoad from 'react-lazyload'
import { handleColors } from '../../utils/handleColors'
import { GridColumnClean } from '../Grid/GridColumn'
import PaddedRow from '../Grid/PaddedRow'

type Props = {
  backgroundImage?: string | null,
  messageBgColor: ?string,
  messageOne: ?string,
  messageTwo: ?string,
}

const Wrap = GridColumnClean.extend`
  background-image: ${({ image }) => `url(${image})`};
  background-size: cover;
  height: 430px;

  @media (min-width: 481px) {
    height: 700px;
  }
`

const Content = styled.div`
  display: grid;
  grid-template-rows: auto 1fr;
`

const MessageRow = PaddedRow.extend`
  align-self: flex-end;
  padding-bottom: 30px;

  @media (min-width: 481px) {
    padding-bottom: 80px;
  }
`

const Message = H1.extend`
  box-sizing: border-box;
  line-height: 1.4;
  margin: 0;

  span {
    background-color: ${({ bgColor, theme }) =>
      bgColor || theme.colors.radicalRed};
    -webkit-box-decoration-break: clone;
    box-decoration-break: clone;
    color: ${({ bgColor, theme }) => theme.contrast(bgColor)};
    padding: 0.2rem 0.6rem;
  }

  &:not(:last-of-type) {
    margin-bottom: 5px;
  }

  @media (min-width: 1024px) {
    width: auto;
  }
`

const Header = ({
  backgroundImage,
  messageBgColor,
  messageOne,
  messageTwo,
}: Props) => {
  const backgroundColor = handleColors(messageBgColor)

  return (
    <LazyLoad height={700} once>
      <Wrap image={backgroundImage}>
        <Content>
          <Navigation />
          <MessageRow>
            {messageOne && (
              <Message bgColor={backgroundColor}>
                <span>{messageOne}</span>
              </Message>
            )}
            {messageTwo && (
              <Message bgColor={backgroundColor}>
                <span>{messageTwo}</span>
              </Message>
            )}
          </MessageRow>
        </Content>
      </Wrap>
    </LazyLoad>
  )
}

export default Header
