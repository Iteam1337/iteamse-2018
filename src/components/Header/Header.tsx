import React from 'react'
import LazyLoad from 'react-lazyload'
import styled, { withProps } from '../../theme'
import { handleColors } from '../../utils/handleColors'
import { GridColumnClean } from '../Grid/GridColumn'
import H1 from '../Typography/H1'
import Navigation from './Navigation'

interface HeaderProps {
  backgroundImage?: string | null
  messageBgColor: string | null
  messageOne: string | null
  messageTwo: string | null
}

interface WrapProps {
  image?: string | null
}

const Wrap = withProps<WrapProps>()(GridColumnClean.extend)`
  background-image: ${({ image }) => `url(${image})`};
  background-size: cover;
  background-position: center;
  height: 430px;

  @media (min-width: 481px) {
    height: 800px;
  }
`

const Content = styled.div`
  background-image: linear-gradient(
    to bottom,
    rgba(0, 0, 0, 0.3),
    rgba(0, 0, 0, 0) 100px
  );
  display: grid;
  grid-column: -1 / 1;
  grid-template-rows: auto 1fr;
  grid-template-columns: 30px 1fr 30px;

  @media (min-width: 1025px) {
    grid-template-columns: 1fr 1024px 1fr;
  }
`

const MessageRow = styled.div`
  align-self: flex-end;
  grid-column: 2;
  grid-row: 2;
  padding-bottom: 30px;

  @media (min-width: 481px) {
    padding-bottom: 80px;
  }
`

interface MessageProps {
  bgColor: string | null
}

const Message = withProps<MessageProps>()(H1.extend)`
  box-sizing: border-box;

  span {
    background-color: ${({ bgColor, theme }) =>
      bgColor || theme.colors.radicalRed};
    -webkit-box-decoration-break: clone;
    box-decoration-break: clone;
    color: ${({ bgColor, theme }) => bgColor && theme.contrast(bgColor)};
    padding: 0 0.6rem;
  }


  @media (min-width: 1025px) {
    width: auto;
  }
`

const Header: React.SFC<HeaderProps> = ({
  backgroundImage,
  messageBgColor = 'red',
  messageOne,
  messageTwo,
}) => {
  const backgroundColor = messageBgColor && handleColors(messageBgColor)

  return (
    <LazyLoad height={700} once>
      <Wrap image={backgroundImage}>
        <Content>
          <Navigation />
          <MessageRow>
            {messageOne && (
              <Message bgColor={backgroundColor}>
                <span>{messageOne}</span>
              </Message>
            )}
            {messageTwo && (
              <Message bgColor={backgroundColor}>
                <span>{messageTwo}</span>
              </Message>
            )}
          </MessageRow>
        </Content>
      </Wrap>
    </LazyLoad>
  )
}

export default Header
