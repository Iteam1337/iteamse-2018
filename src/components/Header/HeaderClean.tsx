import React from 'react'
import { Link } from 'react-router-dom'
import styled from '../../theme'
import { GridColumnClean } from '../Grid/GridColumn'
import logo from './img/iteam_inverted.png'

const LogoLink = styled(Link)`
  &:focus {
    outline: none;
  }
`

const Wrap = styled(GridColumnClean)`
  grid-column: 2;
  grid-row: 1;
`

const Header = styled.div`
  display: grid;
  grid-row: 1;
  width: 150px;
  @media (min-width: 481px) {
    padding: 0;
  }
`

const Logo = styled.img`
  padding: 35px 0;
  max-width: 75%;
  @media (min-width: 1025px) {
    max-width: 100%;
  }
`

const Content = styled.div`
  display: grid;
  grid-column: -1 / 1;
  @media (min-width: 1025px) {
    grid-template-columns: 1fr 1024px 1fr;
  }
`

const HeaderClean = () => {
  return (
    <Wrap>
      <Content>
        <Header>
          <LogoLink to="/">
            <Logo alt="Logo" src={logo} />
          </LogoLink>
        </Header>
      </Content>
    </Wrap>
  )
}

export default HeaderClean
