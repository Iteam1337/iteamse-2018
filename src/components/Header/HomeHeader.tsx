import React from 'react'
import LazyLoad from 'react-lazyload'
import Markdown from 'react-markdown'
import cloudImage from '../../images/cloud.svg'
import greenBlob from '../../images/green-blob.svg'
import styled, { keyframes } from '../../theme'
import { GridColumnClean } from '../Grid/GridColumn'
import H1 from '../Typography/H1'
import iteamI from './img/iteam_i.svg'
import Navigation from './Navigation'

interface HomeHeaderProps {
  title: string
  lead: string
}

const HomeContent = styled.div`
  background-image: url('${greenBlob}');
  background-position: calc(50% - -20px) -130px;
  background-repeat: no-repeat;
  background-size: 560px;
  display: grid;
  grid-column: -1 / 1;
  grid-template-columns: 20px 1fr 20px;
  grid-template-rows: auto 1fr;

  @media (min-width: 1025px) {
    background-position: calc(50% - -220px) -170px;
    background-size: 960px;
    grid-template-columns: 1fr 1024px 1fr;
  }
`

export const HomeMessageRow = styled.div`
  align-items: center;
  display: flex;
  flex-direction: column-reverse;
  grid-column: 2;
  justify-content: center;

  @media (min-width: 481px) {
    flex-direction: row;
    padding: 80px 0;
  }
`

const IllustrationWrap = styled.div`
  align-items: flex-start;
  display: flex;
  flex-basis: 40%;
  justify-content: center;
  margin-top: -60px;
  transform: scale(0.6, 0.6);

  @media (min-width: 1025px) {
    align-items: center;
    justify-content: flex-end;
    margin: 0;
    transform: scale(1, 1);
  }
`

const float = keyframes`
  0% {
    transform: translatey(0px);
  }
  50% {
    transform: translatey(-20px);
  }
  100% {
    transform: translatey(0px);
  }
`

const Cloud = styled.span`
  animation: ${float} 6s ease-in-out infinite;
  background-image: url(${cloudImage});
  background-repeat: no-repeat;
  background-size: contain;
  height: 76px;
  position: absolute;
  right: -80px;
  top: 20px;
  transform: scale(-1, 1);
  width: 136px;
`

const SmallCloud = styled(Cloud)`
  animation-direction: reverse;
  left: 40px;
  top: -60px;
  transform: scale(1, 1);
  height: 53px;
  width: 96px;
`

const IllustrationImage = styled.img`
  height: auto;
  margin: 0;
  width: 100%;
`

const IntroWrap = styled.div`
  flex-basis: 60%;
  z-index: 1;
`

const HomeH1 = styled(H1)`
  font-size: 24px;
  font-weight: 300;
  line-height: 40px;
  margin-bottom: 32px;

  @media (min-width: 1025px) {
    font-size: 30px;
    margin-bottom: 64px;
  }
`

const HomeLead = styled.div`
  font-size: 18px;
  line-height: 1.5;

  @media (min-width: 481px) {
    font-size: 20px;
    font-weight: 300;
    line-height: 24px;
  }

  a {
    border-bottom: 1px solid #000;
    color: inherit;
    text-decoration: none;
  }
`

const HomeHeader: React.SFC<HomeHeaderProps> = ({ title, lead }) => {
  return (
    <>
      <LazyLoad height={700} once>
        <GridColumnClean>
          <HomeContent>
            <Navigation isInverted noShadow />
            <HomeMessageRow>
              <IntroWrap>
                <HomeH1>{title}</HomeH1>
                <HomeLead>
                  <Markdown source={lead} />
                </HomeLead>
              </IntroWrap>
              <IllustrationWrap>
                <Cloud />
                <SmallCloud />
                <IllustrationImage src={iteamI} />
              </IllustrationWrap>
            </HomeMessageRow>
          </HomeContent>
        </GridColumnClean>
      </LazyLoad>
    </>
  )
}

export default HomeHeader
