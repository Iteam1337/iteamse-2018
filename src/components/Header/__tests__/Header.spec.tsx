import * as React from 'react'
import { render, wait } from 'react-testing-library'
import MockedQuery from '../../../utils/test-utils/MockedQuery'
import Header from '../Header'

describe('components/Header', () => {
  const mocks = {}

  it('renders Header', async () => {
    const { container } = render(
      <MockedQuery mocks={mocks}>
        <Header
          backgroundImage={null}
          messageBgColor={null}
          messageOne="Vi digitaliserar fÃ¶retag och organisationer"
          messageTwo="genom strategi, kod och kultur"
        />
      </MockedQuery>
    )

    await wait()

    expect(container).toMatchSnapshot()
  })
})
