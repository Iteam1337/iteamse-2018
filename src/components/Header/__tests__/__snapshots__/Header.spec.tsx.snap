// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Header renders Header 1`] = `
<div>
  <div>
    <div
      class="sc-kIPQKe ftKOiW"
    >
      <div
        class="sc-eXEjpC iOEOtv"
      >
        <div
          class="sc-gzOgki iPgKNP"
        >
          <a
            class="sc-iyvyFf gbVIVA"
          >
            <img
              class="sc-hwwEjo doGCkR"
              data-testid="logo"
              path="test-file-stub"
            />
          </a>
          <div
            class="sc-kPVwWT iGguRZ"
          >
            <a
              class="sc-esjQYD eLRVk"
              aria-current="false"
            >
              Erbjudanden
            </a>
            <a
              class="sc-esjQYD eLRVk"
              aria-current="false"
            >
              Case
            </a>
            <a
              class="sc-esjQYD eLRVk"
              aria-current="false"
            >
              Metod
            </a>
            <a
              class="sc-esjQYD eLRVk"
              aria-current="false"
            >
              Medarbetare
            </a>
            <a
              class="sc-esjQYD eLRVk"
              aria-current="false"
            >
              Karriär
            </a>
            <a
              class="sc-esjQYD eLRVk"
              aria-current="false"
            >
              Om
            </a>
            <div
              class="sc-kfGgVZ hzXCUz"
            />
          </div>
        </div>
        <div
          class="sc-ibxdXY gGaQMA"
        >
          <h1
            class="sc-RefOD kXWznh"
          >
            <span>
              Vi digitaliserar företag och organisationer
            </span>
          </h1>
          <h1
            class="sc-RefOD kXWznh"
          >
            <span>
              genom strategi, kod och kultur
            </span>
          </h1>
        </div>
      </div>
    </div>
  </div>
</div>
`;
